ksql-engine/src/main/java/io/confluent/ksql/internal/KsqlEngineMetrics.java

```diff
@@ -231,4 +261,44 @@ private Sensor createSensor(final Metrics metrics, final String sensorName) {
     return sensor;
   }
 
+
+  private CountMetric configureNumActiveQueriesForGivenState(final Metrics metrics,
+      final KafkaStreams.State state) {
+    final String gaugeName = ksqlServiceId + metricGroupName + "-" + state + "-queries";
+    final Gauge<Long> gauge = new Gauge<Long>() {
+      @Override
+      public Long value(final MetricConfig metricConfig, final long l) {
+        return ksqlEngine.getPersistentQueries()
+            .stream()
+            .filter(queryMetadata -> queryMetadata.getKafkaStreams().state() == state)
+            .count();
+      }
+    };
+
+    final MetricName metricName = metrics.metricName(gaugeName,
+        metricGroupName,
+        String.format("Count of queries in %s state.", state.toString()));
+    final CountMetric countMetric = new CountMetric(metricName, gauge);
+    metrics.addMetric(metricName, gauge);
+    countMetrics.add(countMetric);
+    return countMetric;
+  }
+
+  private static class CountMetric {
```

Comment:
nit: move out into own class, rather than nested.

Reply:
This is only being used here. Any specific reason to make it it's own class?

Reply:
To reduce the complexity of this class.